test/v3/controllers/requestParsers/validators/SubmitSelfEmploymentBsasValidatorSpec.scala

```diff
@@ -57,117 +61,127 @@ class SubmitSelfEmploymentBsasValidatorSpec extends UnitSpec {
       }
     }
-    "return a single error" when {
-      "passed an invalid nino" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData("beans", bsasId, AnyContentAsJson(Json.toJson(defaultBody())))) shouldBe List(NinoFormatError)
+
+    "return NinoFormatError error" when {
+      "an invalid nino is supplied" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(nino = "A12344A", calculationId = calculationId, AnyContentAsJson(mtdRequest))) shouldBe
+          List(NinoFormatError)
       }
-      "passed an invalid bsasId" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, "beans", AnyContentAsJson(Json.toJson(defaultBody())))) shouldBe List(CalculationIdFormatError)
+    }
+
+    "return CalculationIdFormatError error" when {
+      "an invalid calculationId is supplied" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(nino = nino, calculationId = "12345", AnyContentAsJson(mtdRequest))) shouldBe
+          List(CalculationIdFormatError)
       }
-      "passed an empty body" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId, AnyContentAsJson(Json.obj()))) shouldBe List(RuleIncorrectOrEmptyBodyError)
+    }
+
+    "return RuleIncorrectOrEmptyBodyError" when {
+      "an empty body is submitted" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(nino = nino, calculationId = calculationId, AnyContentAsJson(Json.obj()))) shouldBe List(
+          RuleIncorrectOrEmptyBodyError)
       }
-      "passed an invalid body" in {
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj("income" -> "beans")))) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj("income" -> "{}")))) shouldBe List(RuleIncorrectOrEmptyBodyError.copy(paths = Some(Seq("/income"))))
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.toJson(defaultBody(income = Some(Income(None, None))))))) shouldBe List(RuleIncorrectOrEmptyBodyError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(Json.obj()))) shouldBe List(RuleIncorrectOrEmptyBodyError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(mtdRequestWithBothExpenses))) shouldBe List(RuleBothExpensesError)
-        validator.validate(SubmitSelfEmploymentBsasRawData(nino, bsasId,
-          AnyContentAsJson(mtdRequestWithAdditionsAndExpenses))) shouldBe List(RuleBothExpensesError)
+
+      "object is empty" in {
+        validator.validate(SubmitSelfEmploymentBsasRawData(
+          nino,
+          calculationId,
+          AnyContentAsJson(Json.obj("income" -> "{}")))
```

Comment:
This results in JSON:
``` json
{
  "income": "{}"
}
```
i.e. income is a string value not an empty object.  

Suspect it should be:
```suggestion
          AnyContentAsJson(Json.obj("income" -> JsObject.empty)))
```